breathecode/events/views.py

```diff
@@ -725,6 +725,58 @@ def get(self, request, event_type_slug, academy_id=None):
         serializer = EventTypeVisibilitySettingSerializer(items, many=True)
         return handler.response(serializer.data)
 
+    @capable_of('crud_event_type')
+    def post(self, request, event_type_slug, academy_id=None):
+        handler = self.extensions(request)
+
+        event_type = EventType.objects.filter(slug=event_type_slug).first()
+        if not event_type:
+            raise ValidationException('Event type not found', slug='event-type-not-found')
+
+        academy = Academy.objects.filter(id=request.data['academy']).first()
+        if not academy:
+            raise ValidationException('Academy not found', slug='academy-not-found')
+
+        syllabus = None
+        if 'syllabus' in request.data:
+            syllabus = Syllabus.objects.filter(id=request.data['syllabus']).first()
+
+        cohort = None
+        if 'cohort' in request.data:
+            cohort = Cohort.objects.filter(id=request.data['cohort']).first()
+
+        visibility_setting, created = EventTypeVisibilitySetting.objects.get_or_create(syllabus=syllabus,
+                                                                                       academy=academy,
+                                                                                       cohort=cohort)
+
+        serializer = EventTypeVisibilitySettingSerializer(visibility_setting, many=False)
+        return Response(serializer.data, status=status.HTTP_201_CREATED if created else status.HTTP_200_OK)
+
+    @capable_of('crud_event_type')
+    def delete(self, request, event_type_slug, visibility_setting_id=None, academy_id=None):
+        item = EventTypeVisibilitySetting.objects.filter(id=visibility_setting_id).first()
+
+        if not item:
+            raise ValidationException('Event type visibility setting not found',
+                                      404,
+                                      slug='event-type-visibility-setting-not-found')
+
+        event_type = EventType.objects.filter(slug=event_type_slug).first()
+        if not event_type:
+            raise ValidationException('Event type not found', slug='event-type-not-found')
+
+        other_event_type = EventType.objects.filter(visibility_settings__id=visibility_setting_id).exclude(
```

Comment:
Delete this visibility just in this `EventType`, a academy is not able to edit the visibility setting of another academy, if no one `EventType` of any academy is using this visibilitysetting, this must be removed